cloud/azure/deploy/website.go

```diff
@@ -0,0 +1,520 @@
+// Copyright Nitric Pty Ltd.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package deploy
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io/fs"
+	"log"
+	"mime"
+	"net/url"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/cdn/armcdn"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/nitrictech/nitric/cloud/common/deploy/pulumix"
+	deploymentspb "github.com/nitrictech/nitric/core/pkg/proto/deployments/v1"
+	"github.com/samber/lo"
+
+	cdn "github.com/pulumi/pulumi-azure-native-sdk/cdn/v2"
+	"github.com/pulumi/pulumi-azure-native-sdk/storage"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+func (p *NitricAzurePulumiProvider) createStaticWebsite(ctx *pulumi.Context, resources []*pulumix.NitricPulumiResource[any]) error {
+	var rootWebsite *deploymentspb.Website
+	var err error
+
+	for _, resource := range resources {
+		config, ok := resource.Config.(*deploymentspb.Resource_Website)
+
+		if ok && config.Website.BasePath == "/" {
+			rootWebsite = config.Website
+			break
+		}
+	}
+
+	if rootWebsite == nil {
+		return fmt.Errorf("no root website configuration found")
+	}
+
```

Comment:
does this check also exist in the AWS implementation?

Reply:
I'll move this to the CLI